tests/app/dao/test_templates_dao.py

```diff
@@ -957,3 +959,67 @@ def test_template_with_provider_id_persists_provider_id(
     assert notify_db_session.session.get(Template, template.id).provider_id == provider.id
     # Teardown
     template_cleanup(notify_db_session.session, template.id)
+
+
+class TestDAOGetTemplateById:
+    def test_dao_get_template_by_id_with_no_version(
+        self,
+        sample_template: Callable[..., Any],
+    ):
+        template = sample_template()
+
+        result = dao_get_template_by_id(template.id)
+
+        assert result == template
+        assert isinstance(result, Template)
+        assert result.id == template.id
+        assert result.version == 1
```

Comment:
I think everything after the first assertion is redundant.
```suggestion
        assert result == template
```
Check Pep8, but you also might want that to be `assert result is template`.  I think that indicates they point to the same location in memory.